source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,198 @@
+    { "_id": 2, "color": "green", "qty": 8 }
+    { "_id": 3, "color": "purple", "qty": 4 }
+    { "_id": 4, "color": "green", "qty": 11 }
+
+Read operations
+---------------
+
+.. _retrieve-find:
+
+Find
+~~~~
+
+Use the find operation to retrieve a subset of your existing data in
+MongoDB. To further specify which documents to retrieve, you can sort,
+skip, and limit the documents or fields.
+
+To perform a find operation, call the ``find()`` method on an instance
+of a ``MongoCollection``. This method searches a collection for documents that
+match the query filter you provide. For more information on how to
+specify a query, see our :doc:`Specify a Query
+</fundamentals/crud/query-document>` guide. 
+
+To access the results, you can iterate through the results using a
+cursor. For more information on query filters, see our :doc:`Access Data
+From a Cursor </fundamentals/crud/read-operations/cursor>` guide. 
+
+Example
+```````
+
+The paint store would like to know which orders contain more than three, but
+less than ten cans of paint.
+
+To address this scenario, the paint store needs to find orders where the
+quantity of paint ordered is greater than "3", but less than "10" : 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Retrieve.java
+   :language: java
+   :dedent:
+   :start-after: begin findExample
+   :end-before: end findExample
+
+For more information how to build filters, see our :doc:`Filters Builders
+</fundamentals/builders/filters>` guide.
+
+The following shows the output of the query specified above:
+
+.. code-block:: json
+
+    { "_id": 2, "color": "green", "qty": 8 }
+    { "_id": 3, "color": "purple", "qty": 4 }
+
+After the paint store runs this query, they find that orders with
+``_id`` "2" and "3" contain more than three, but less than ten cans
+of paint.
+
+For a runnable ``find()`` example, see our :doc:`Find Multiple
+Documents </usage-examples/find>` page. 
+
+.. _retrieve-aggregate:
+
+Aggregate
+~~~~~~~~~
+
+Use the aggregate operation to perform the stages in an aggregation
+pipeline.  An aggregation pipeline is a multi-staged transformation that
+produces an aggregated result. 
+
+To perform an aggregate operation, call the ``aggregate()`` method on an
+instance of a ``MongoCollection``. This method accepts aggregation
+expressions to run in sequence. To perform transformations, you can
+define aggregation stages that specify how to match documents, rename
+fields and group values. For more information on aggregation, see our
+:doc:`Aggregation  </fundamentals/aggregation>` guide.
+
+Example
+```````
+
+The paint store would like to know which color is the most purchased
+(highest quantity sold).
+
+To address the scenario, the paint store needs to create an aggregation
+pipeline that:
+
+- Matches all the documents in the ``paint_order`` collection
+- Groups orders by colors 
+- Sums up the quantity field by color
+- Sorts the results in descending order by quantity
```

Comment:
Nitpick:
I think "descending order by quantity" might be slightly confusing. This might be a better way to phrase this item:
```suggestion
- Orders the results by highest-to-lowest quantity
```